ipatests/test_integration/test_trust.py

```diff
@@ -993,3 +1019,154 @@ def test_server_option_with_unreachable_ad(self):
+    """
+    topology = 'line'
+    num_ad_domains = 1
+    num_clients = 1
+
+    def test_add_posix_trust(self):
+        tasks.configure_dns_for_trust(self.master, self.ad)
+        tasks.establish_trust_with_ad(
+            self.master, self.ad_domain,
+            extra_args=['--range-type', 'ipa-ad-trust-posix'])
+
+    @pytest.mark.parametrize('type', ['hybrid', 'true', "false"])
+    def test_gidnumber_not_coresponding_existing_group(self, type):
+        """
+        Test checks that AD users which contain posix attributes
+        (uidNumber and gidNumber) but there is no group with the
+        corresponding gidNumber.
+        """
+        posixuser = "testuser2@%s" % self.ad_domain
+        self.mod_idrange_auto_private_group(type)
+        (uid, gid) = self.get_user_id(self.master, posixuser)
+        if type == "true":
+            assert uid == gid
+        else:
+            assert uid != gid
+
+    @pytest.mark.parametrize('type', ['hybrid', 'true', "false"])
+    def test_only_uid_number_auto_private_group_default(self, type):
+        """
+        Test checks that posix user with only uidNumber defined
+        and gidNumber not set, auto-private-group
+        is set to false/true/hybrid
+        """
+        posixuser = "testuser1@%s" % self.ad_domain
+        self.mod_idrange_auto_private_group(type)
+        if type == "true":
+            (uid, gid) = self.get_user_id(self.clients[0], posixuser)
+            assert uid == gid
+        else:
+            for host in [self.master, self.clients[0]]:
+                result = host.run_command(['id', posixuser], raiseonerr=False)
+                tasks.assert_error(result, "no such user")
+
+    @pytest.mark.parametrize('type', ['hybrid', 'true', "false"])
+    def test_auto_private_group_primary_group(self, type):
+        """
+        Test checks that AD users which contain posix attributes
+        (uidNumber and gidNumber) and there is primary group
+        with gid number defined..
+        """
+        posixuser = "testuser@%s" % self.ad_domain
+        self.mod_idrange_auto_private_group(type)
+        (uid, gid) = self.get_user_id(self.clients[0], posixuser)
+        if type == "true":
+            assert uid == gid
+
+    @pytest.mark.parametrize('type', ['hybrid', 'true', "false"])
+    def test_idoverride_with_auto_private_group(self, type):
+        """
+        Override ad trusted user in default trust view
+        and set auto-private-groups=[hybrid,true,false]
+        and ensure that overridden values takes effect.
+        """
+        test_uid = "99999999"
+        test_gid = "99999999"
+        posixuser = "testuser@%s" % self.ad_domain
+        result = self.master.run_command(
+            ["ipa", "idoverrideuser-add", "Default Trust View",
+             "--uid", test_uid, "--gid", test_gid, posixuser],
+            raiseonerr=False)
+        if result.returncode != 0:
+            tasks.assert_error(result, "already exists")
```

Comment:
It's better to use a fixture that adds and removes the idoverrideuser, to avoid polluting other tests.